### PR TITLE
added functionality to store username in persistent storage

### LINKED ISSUE #8: Store username in a persistent storage
**Note:** This issue is reserved for SLoP Contributors.

Can use [shared_preferences](https://pub.dev/packages/shared_preferences)

### PR DESCRIPTION
**Pull request by GajjarMiten**
## Description

used shared_prefrences for storage in retrieving username from local storage. 

## Tests
I added the following tests:
- added unit test in Storage_service_test.dart
- added unit test in Store_test.dart 

## Related Issues

Fixes  #8

### CODE REVIEW

File: test/unit test/Storage_service_test.dart [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import 'package:sinix_android/services/Storage_service.dart';
+
+void main() {
+  test('Local Storage test', () async {
+    final LocalStorage localStorage = LocalStorage();
+
+    await localStorage.init();
+
+    // firstOpen will always be true in test
+    expect(localStorage.firstOpen, true);
```
Comment (sanket143) [inline test/unit test/Storage_service_test.dart:15]: This test is failing for me. Can you try this out once more and update.

File: test/widget_test.dart [hunk 2]
```diff
@@ -8,7 +8,7 @@
 import 'package:flutter/material.dart';
```
Comment (sanket143) [inline test/widget_test.dart]: I think we can get rid of this file since it isn't making any contributiong.

**Review by sanket143 — CHANGES_REQUESTED** [review]

**Comment by sanket143** [comment]
I liked that you added tests. I've updated the points to 75 from 25.

**Comment by GajjarMiten** [comment]
Hey @sanket143!

I have updated the test files now it should pass all the tests :)

File: test/unit test/Storage_service_test.dart (hunk 1 above)
Comment (GajjarMiten) [reply test/unit test/Storage_service_test.dart:15]: Yes, there was a mistake by me.
actually, I didn't know that we have to initialize shared preferences with mock data when we are testing it.

**Review by sanket143 — APPROVED** [review]

**Comment by sanket143** [comment]
give @GajjarMiten 75 points

**Pull request merged**